Mathlib/Algebra/Polynomial/Bivariate.lean

```diff
@@ -197,6 +197,51 @@ lemma eval_C_X_eval₂_map_C_X {p : R[X][Y]} :
 
 end
 
+section aevalAeval
+
+noncomputable section
+
+variable {R A : Type*} [CommRing R] [CommRing A] [Algebra R A]
+
+/-- `aevalAeval x y` is the `R`-algebra evaluation morphism sending a two-variable polynomial
+  `p : R[X][Y]` to `p(x,y)`. -/
+abbrev aevalAeval (x y : A) : R[X][Y] →ₐ[R] A :=
+  ((aeval x).restrictScalars R).comp
+    (letI := Polynomial.algebra; (aeval (R := R[X]) (C y)).restrictScalars R)
+
+theorem coe_aevalAeval_eq_evalEval (x y : A) : ⇑(aevalAeval x y) = evalEval x y := by
+  ext p; simp [aevalAeval, evalEval, aeval, eval, Algebra.ofId]
+
+/-- The R-algebra automorphism given by `X ↦ Y` and `Y ↦ X`. -/
+def Bivariate.swap : R[X][Y] ≃ₐ[R] R[X][Y] := by
+  apply AlgEquiv.ofAlgHom (aevalAeval (Y : R[X][Y]) (C X)) (aevalAeval (Y : R[X][Y]) (C X))
+    <;> (ext n m <;> simp)
+
+/-- Evaluating `swap p` at `x`, `y` is the same as evaluating `p` at `y` `x`. -/
+theorem Bivariate.aevalAeval_swap (x y : A) (p : R[X][Y]) :
+    aevalAeval x y (swap p) = aevalAeval y x p := by
+  unfold swap aevalAeval at *
```

Comment:
This `unfold` line tells me that we are missing some `apply` lemmas, can you please add them?